File: Palindrome with minimum sum.cpp
class Solution {
  public:
    int minimumSum(string s) {
    // code here
    int lo=0,hi=s.length()-1;
    while(lo < hi){
        if(s[lo] != s[hi]){
            if(s[lo] == '?')s[lo] = s[hi];
            else if(s[hi] == '?')s[hi] = s[lo];
            else return -1;
        }
        lo++;
        hi--;
    }
    
    int cnt = 0,prev = -1;
    for(const auto& x: s){
        if(isalpha(x)){
            if(prev != -1)cnt += abs(prev-x);
            prev = x;
        }
    }
    
    return cnt;
}
};

File: Palindrome with minimum sum.java
class Solution {
    public static int minimumSum(String s) {
        // code here
        char d[] = s.toCharArray();
         int n=s.length();
        int i=0,j=n-1;
        int count=0;
        char prev='$';
        while(i<j)
        {
            if(d[i]!='?' && d[j]!='?' && d[i]!=d[j]) return -1;
            else if(d[i]=='?' && d[j]!='?') 
            {
                // s[i]=s[j];
                if(prev!='$') count+=Math.abs(d[j]-prev);
                prev=d[j];
            }
            else if(d[i]!='?' && d[j]=='?') 
            {
                // s[j]=s[i];
                if(prev!='$') count+=Math.abs(d[i]-prev);
                prev=d[i];    
            }
            else if(d[i]==d[j] && d[i]!='?')
            {
                if(prev!='$') count+=Math.abs(d[i]-prev);
                prev=d[i];  
            }
            i++;j--;
        }
        return 2*count;
    }
}
